frontend/src/Crossword.tsx

```diff
@@ -364,10 +383,41 @@ const Crossword: React.FC<CrosswordProps> = (props) => {
     // @ts-ignore
     onExternalWrite.current = handleExternalWrite;
 
+    const handleTouchStart = (event: TouchEvent) => {
+      if (event.touches.length > 1) {
+        // Start panning
+        if (isDrawing) {
+          stopDrawing();
+        }
+        return;
+      }
+      const [x, y] = getTouchLocation(event);
+      startDrawing(x, y);
+    };
+
+    const handleTouchMove = (event: TouchEvent) => {
+      if (event.touches.length > 1 || !isDrawing) {
+        // Panning
+        return;
+      }
+      const [x, y] = getTouchLocation(event);
+      draw(x, y);
+      event.preventDefault();
+    };
+
+    const handleTouchEnd = (event: TouchEvent) => {
+      if (isDrawing && event.touches.length === 0) {
+        stopDrawing();
+      }
+    };
+
     window.onkeydown = handleKey;
-    canvas.onmousedown = (event) => startDrawing(event);
-    canvas.onmousemove = draw;
-    canvas.onmouseup = stopDrawing;
+    canvas.onmousedown = handleMouseDown;
+    canvas.onmousemove = handleMouseMove;
+    canvas.onmouseup = handleMouseUp;
+    canvas.ontouchstart = handleTouchStart;
```

Comment:
Implementing the touch stuff like this got the touch drawing working for me :)
```
canvas.addEventListener("touchstart", handleTouchStart, false)
```

Reply:
Works fine for me as is

Reply:
Also adding this CSS property to the drawing layer fixes panning and zooming.
```
touch-action: pinch-zoom;
```
(https://css-tricks.com/almanac/properties/t/touch-action/#article-header-id-1)

Reply:
> Works fine for me as is

RIP HP EliteBook touchscreen :D

Reply:
Ah, I didn’t know about that CSS property! I think that may be exactly what we’re looking for here. I’ll try that out tomorrow.

Reply:
What browser were you running on the EliteBook?